app/public/js/common/configLocation.js

```diff
@@ -38,12 +38,26 @@ const configuration = {
       userConfigPath = `${userHome}/Library/Preferences/Soundnode`;
     }
 
+    // Guard to assert type of string.
+    if (typeof userConfigPath !== "string") {
+      throw `Could not set userConfigPath for this OS ${process.platform}`
+    }
+
     // create user config in path
     // if there is no userConfig path
-    if (!fs.statSync(userConfigPath).isDirectory()) {
-      this.createUserConfig()
+    //
+    // fs.statSync will throw an exception if
+    // any directory along the path doesn't exist.
+    // Solution: use try catch.
+    try {
+      var fi = fs.statSync(userConfigPath);
```

Comment:
can you rename the variable to something more clear please?

Reply:
Given the tiny scope of the variable I felt the name to be adequate.
What would you consider to be a more clear name? Maybe something like `configFileInfo`?

Reply:
`configFileInfo` would be more explicit,

Since the variable is never mutated, could you change it to a constant?